classes/publication/Repository.inc.php

```diff
@@ -147,9 +151,12 @@ public function version(Publication $publication): int
         }
 
         // Chapters (and all associated objects)
-        $chapterAuthorDao = DAORegistry::getDAO('ChapterAuthorDAO'); /** @var ChapterAuthorDAO $chapterAuthorDao */
         $newAuthors = iterator_to_array(
-            Services::get('author')->getMany(['publicationIds' => $newPublication->getId()])
+            Repo::author()->getMany(
+                Repo::author()
+                    ->getCollector()
+                    ->filterByPublicationIds([$newPublication->getId()])
+            )
```

Comment:
Same thing here. You don't need the `iterator_to_array` call now that it's returning a `LazyCollection`.

Reply:
Changed.